### DISCUSSION TITLE
Optimizing material variety

### DISCUSSION THREAD

**Discussion by maxfrei750** [mitsuba-renderer/mitsuba3#713] [Q&A] [answered]
I'd like to create a set of objects, which share one common material, but each feature small variations with respect to the individual material parameters (e.g. following a normal distribution). Later, I'd like to tune these parameter variations (e.g. the mean and standard deviation of the underlying normal distribution) using optimization.

As far as I understand, a way to achieve this would be:
1. Create individual objects, with identical, yet separate materials. 
2. Define latent variables (e.g. for the mean and standard deviation).
3. Link these variables to the material properties, by sampling from the normal distribution.

Is there anything in the way with this approach?

Thanks!

EDIT: This question is somewhat similar to https://github.com/mitsuba-renderer/mitsuba3/discussions/566, I'm just interested if there are for instance special considerations w.r.t. to shaders and latent variables.

**Comment by njroussel** [comment, marked as answer]
Hi @maxfrei750 

Looks all good to me! I have no idea how well this might actually work, but there shouldn't be any specific Mitsuba/Drjit hurdles.

**Reply by maxfrei750** [reply]
I'm glad to hear that. Thanks!
